include/aws/http/request_response.h

```diff
@@ -939,6 +941,12 @@ int aws_http_message_erase_header(struct aws_http_message *message, size_t index
  *
  * Tip for language bindings: Do not bind the `options` struct. Use something more natural for your language,
  * such as Builder Pattern in Java, or Python's ability to take many optional arguments by name.
+ *
+ * Note: The header of the request will be sent as it is. Unless HTTP/1 message used on HTTP/2 connection, then
```

Comment:
nit: this is a bit confusing. maybe explicitly state that for http1 messages on http1 connection and http2 on http2 con headers are sent as is, but for corner case of http1 message on http2 this corner case applies. 

Might also be worth mentioning what happens with reverse http2 message on http1 conn